src/wp-includes/blocks.php

```diff
@@ -1056,9 +1056,64 @@ function apply_block_hooks_to_content( $content, $context, $callback = 'insert_h
 		$after_block_visitor  = make_after_block_visitor( $hooked_blocks, $context, $callback );
 	}
 
-	$blocks = parse_blocks( $content );
+	$block_allows_multiple_instances = array();
+	/*
+	 * Remove hooked blocks from `$hooked_block_types` if they have `multiple` set to false and
+	 * are already present in `$content`.
+	 */
+	foreach ( $hooked_blocks as &$relative_positions ) {
+		foreach ( $relative_positions as &$hooked_block_types ) {
+			foreach ( $hooked_block_types as &$hooked_block_type ) {
+				$hooked_block_type_definition =
+					WP_Block_Type_Registry::get_instance()->get_registered( $hooked_block_type );
+				if ( block_has_support( $hooked_block_type_definition, 'multiple', true ) ) {
+					$block_allows_multiple_instances[ $hooked_block_type ] = true;
+				} elseif ( has_block( $hooked_block_type, $content ) ) {
+					unset( $hooked_block_type );
+					if ( empty( $hooked_block_types ) ) {
+						unset( $hooked_block_types );
+					}
+					if ( empty( $relative_positions ) ) {
+						unset( $relative_positions );
+					}
```

Comment:
This needs to be double-checked. It seems like `unset` on the reference only destroys the local variable here, it does not remove the value from the array.

Reply:
Good spot! Should be fixed per https://github.com/WordPress/wordpress-develop/pull/7443/commits/257225c98c4a4354aa89c841b0663ca91c24c6ca 🤞